src/main/java/sirius/kernel/xml/ContentDispositionHelper.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ *
+ * Based on DownloadUtils by mozilla-mobile/android-components
+ * https://github.com/mozilla-mobile/android-components/blob/main/components/support/utils/src/main/java/mozilla/components/support/utils/DownloadUtils.kt
+ * Right for use and modification was granted under the Mozilla Public License 2.0 , http://mozilla.org/MPL/2.0/
+ */
+
+package sirius.kernel.xml;
+
+import sirius.kernel.commons.Strings;
+import sirius.kernel.health.Exceptions;
+
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Static helper class which helps to parse filenames from content disposition headers
+ */
+public class ContentDispositionHelper {
```

Comment:
_Helper classes have a special semantic in sirius (see sirius-web) - maybe "ContentDispositionParser" or the like?